### ISSUE TITLE
Files to be ignored depend on working directory

### ISSUE THREAD

**Issue by paleite** [eslint/eslint#14775]
I've created a minimal reproduction repo here: https://github.com/paleite/eslintignore-bug

**Tell us about your environment**

Node version: v14.16.1
npm version: v7.12.1
Local ESLint version: v7.30.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 20.5.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
console.log(`Loaded .eslintrc from the root folder. CWD is ${process.cwd()}`);

module.exports = {
  env: { es6: true, jest: true, node: true },
  extends: "eslint:recommended",
  root: true,
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

When running eslint with the same ignore-file from different paths, I expected the ignorefile to behave the same way, but got different results. I used 2 commands:

```bash
eslint --config $(git rev-parse --show-toplevel)/.eslintrc.js --ignore-path $(git rev-parse --show-toplevel)/.list-of-ignored-files package-a/
```

and

```bash
cd package-a/ && eslint --config $(git rev-parse --show-toplevel)/.eslintrc.js --ignore-path $(git rev-parse --show-toplevel)/.list-of-ignored-files .
```

**What did you expect to happen?**

I expected both commands to have the same result

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

I got different results

**Steps to reproduce this issue:**

I've created a minimal reproduction repo here: https://github.com/paleite/eslintignore-bug
The steps (as detailed in the README.md in that repo) are as follows:

```bash
npm install           # Install modules
npm run lint-absolute # Starts ESLint from the git root folder and uses the ignore-file from the git root folder
npm run lint-relative # Starts ESLint from a subfolder and uses the ignore-file from the git root folder
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

Hopefully this is a good candidate for a new contributor.

**Comment by paleite** [comment]
Related PR, which was supposed to fix above issue back in v4.0.0: https://github.com/eslint/eslint/issues/6759

**Comment by nzakas** [comment]
Thanks for all the research and the repo case. Given that we wanted to fix this earlier, it seems like a good idea to revisit.

I think we once again need to decide if this is a breaking change or not. Given how long the behavior has been this way, I’m leaning towards it being a breaking change but I’d like to hear what others think.

**Comment by paleite** [comment]
Awesome! Thanks 🙂

I was hoping to get @IanVS to chime in, because he was the person who previously merged the original PR and made some really good points, imho.

**Comment by IanVS** [comment]
I haven't been involved in core ESLint dev for a while, but I did check out your repo @paleite, and as far as I can tell, the behavior changed / broke in 7.0.0.  I didn't dig too deeply, but if I had to guess, I'd say it is probably related to https://github.com/eslint/eslint/pull/12887, from the release notes:

>Starting in ESLint v7, configuration files and ignore files passed to ESLint using the --config path/to/a-config and --ignore-path path/to/a-ignore CLI flags, respectively, will resolve from the current working directory rather than the file location. This allows for users to utilize shared plugins without having to install them directly in their project.

**Comment by paleite** [comment]
@IanVS , thanks for looking into it. It’s very valuable to me.

@nzakas , I appreciate your openness to look into this again. I think now might be a good time to decide how this should work, before version 8 becomes stable.

My main reasoning behind why I considered this to be a bug is because it breaks the assumptions of the developer, ie the principle of least astonishment (POLA); in my personal opinion, I expect ESLint to handle its ignorefiles the same way git and others do, but I’m open to arguments why it shouldn’t be like that. 

The release note @IanVS mentioned, said it was using the ignore-file so you utilize shared plugins, but i don’t have enough context to understand the argument that note was trying to make.

Thanks again, guys!

**Comment by nzakas** [comment]
The breaking changes for v8 have already been finalized, so if we decide this is a breaking change, it will have to wait until v9. We are a bit shorthanded right now, so unfortunately it will probably take us a little while to get back to looking at this.

**Comment by btmills** [comment]
I agree this behavior is unexpected. I'd also lean toward this being a breaking change - I wouldn't be surprised if people are relying on this behavior to have a single ignore file shared by multiple sub-projects and expecting the paths to be relative.

**Comment by nzakas** [comment]
@eslint/eslint-tsc how do we want to proceed here?

**Comment by btmills** [comment]
The <7.0.0 way, where paths are relative to the `.eslintignore` file, worked with monorepos. The >=7.0.0 way, where paths are relative to the cwd, [works with shared files](https://github.com/eslint/rfcs/tree/main/designs/2019-changing-base-path-in-config-files-that-cli-options-specify#motivation) from e.g. npm packages. We decided to make the change in RFC37, so we shouldn't take reversing that RFC lightly. Neither do I see a great way to work with both short of adding an option to specify the relative resolution base directory.

**Comment by nzakas** [comment]
Given the tie-in with eslintrc, it seems like a good idea to wait until eslintrc is removed before considering making a change here.

**Comment by nzakas** [comment]
I’ll keep this open to review once the new config system is testable.

**Comment by mdjermanovic** [comment]
Since we're not changing eslintrc, and the new config system does not support ignore files, it looks like we can close this issue?

**Comment by nzakas** [comment]
Agreed. This problem won't exist with the new config system and we aren't changing eslintrc, so it's safe to close this.
